tests/test_irbempy.py

```diff
@@ -290,25 +290,132 @@ def test_GSM_SM_init(self):
-    ##	unittest.TextTestRunner(verbosity=2).run(suite)
+class IRBEMShieldoseTests(unittest.TestCase):
 
-    unittest.main()
+    def setUp(self):
+        super(IRBEMShieldoseTests, self).setUp()
+        self.depths_mil = np.logspace(np.log10(4), np.log10(5000), 30)
+        self.depths_mm = self.depths_mil * 0.0254
+        self.sd_default = ib.Shieldose2()
+
+    def tearDown(self):
+        super(IRBEMShieldoseTests, self).tearDown()
+
+    def test_setshielding(self):
+        """Setting shielding should appropriately update settings
+        """
+        sdd = self.sd_default
+        tunit = 'mm'
+        target = np.logspace(0, 3, 50)
+        tcopy = target.copy()
+        sdd.set_shielding(depths=target, units=tunit)
+        target *= 2  # make sure we're not modifying arrays
+        curr_dep = np.asarray(sdd['settings']['depths'].copy())
+        numpy.testing.assert_array_almost_equal(tcopy, curr_dep)
+        numpy.testing.assert_string_equal(sdd['settings']['depths'].attrs['UNITS'], tunit)
+
+    def test_mil_mm(self):
+        """Should get same results for different depth units
+        """
+        sd1 = self.sd_default
+        sd1.set_shielding(depths=self.depths_mil, units='Mil')
+        sd1.get_dose()
+        sd2 = ib.Shieldose2()
+        sd2.set_shielding(depths=self.depths_mm, units='mm')
+        sd2.get_dose()
+        de1 = np.asarray(sd1['results']['dose_electron'])
+        de2 = np.asarray(sd2['results']['dose_electron'])
+        numpy.testing.assert_array_almost_equal(de1, de2)
+
+    def test_shieldose2_changed(self):
+        """Results should be removed if settings are changed
+        """
+        self.assertFalse('results' in self.sd_default)
+        self.sd_default.get_dose()
+        self.assertTrue('results' in self.sd_default)
+        self.sd_default.set_shielding(depths=self.depths_mm, units='mm')
+        self.assertFalse('results' in self.sd_default)
+
+    def test_invalid_det(self):
+        """Asking for invalid detector material raises ValueError
+        """
+        self.assertRaises(ValueError, self.sd_default.get_dose, detector=99)
+
+    def test_invalid_fluence(self):
+        """Asking for invalid fluence model issues UserWarning
+        """
+        self.assertWarns(UserWarning,
+                         self.sd_default.get_dose,
+                         fluence='notamodel')
+
+    def test_e_j_mismatch(self):
+        """Mismatched flux energy array sizes raises ValueError
+        """
+        jarr = np.logspace(0, 4, 35)[::-1]
+        earr = np.logspace(1, 10, 40)
+        self.assertRaises(ValueError, self.sd_default.set_flux,
+                          jarr, earr, 'e')
+
+    def test_plot_e(self):
+        """Check for expected outputs in electron dose plot"""
+        self.sd_default.get_dose()
+        res = self.sd_default['results']
+        f1, a1, l1 = self.sd_default.plot_dose(source=['e'])
+        f2, a2, l2 = self.sd_default.plot_dose(source=['e', 'brems'])
```

Comment:
See spacepy_testing.TestPlot, which makes sure we don't need an X server running. Can we just plug that in?

Also there's a fair bit of the other options not tested--just exhaustion? It feels like the kind of thing that might bite us and I can probably dump something out.

Reply:
I can add a similar test for the protons, although this was mostly to test that the plot correctly combines the direct electron dose with the bremsstrahlung dose, since they're separate outputs from the fortran code.

Reply:
Turns out the Bremsstrahlung-only case was broken once I tested it, so that's fixed now too.